# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/NeckoParent.cpp
# Commit: bfe21f274ed5
# Full Hash: bfe21f274ed5a4630a3503bcb875fed04f90533c
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-04-16 03:00:04
# Regressor Bug: 1594529
# File Overlap Count: 1
# Description:
#   Bug 1594529 - Create LoadInfo for subdocuments directly in parent process with DocumentChannel. r=mattwoodrow,nika
#   
#   Currently, with Fission enabled we are not able to create a proper LoadInfo
#   object when doing a subdocument load because we do not have access to a loading
#   context if the load is happening inside of an OOP frame. To solve this problem,
# ==============================================================================

diff -r 29ff609063c2 -r bfe21f274ed5 netwerk/ipc/NeckoParent.cpp
--- a/netwerk/ipc/NeckoParent.cpp	Wed Apr 15 18:53:22 2020 +0000
+++ b/netwerk/ipc/NeckoParent.cpp	Wed Apr 15 18:53:06 2020 +0000
@@ -420,8 +420,16 @@
     context = aContext.get_canonical();
   }
 
-  nsCOMPtr<nsIPrincipal> requestingPrincipal =
-      GetRequestingPrincipal(Some(args.loadInfo()));
+  nsCOMPtr<nsIPrincipal> requestingPrincipal;
+  // We only have the requesting principal in case of TYPE_SUBDOCUMENT.
+  // If we don't have an embedder window global, then it is probably a race and
+  // we will deal with that later in the code path.
+  if (context && !aContext.IsDiscarded() && context->GetParent()) {
+    if (RefPtr<WindowGlobalParent> embedderWGP =
+            context->GetParentWindowGlobal()) {
+      requestingPrincipal = embedderWGP->DocumentPrincipal();
+    }
+  }
 
   nsCOMPtr<nsILoadContext> loadContext;
   const char* error = CreateChannelLoadContext(